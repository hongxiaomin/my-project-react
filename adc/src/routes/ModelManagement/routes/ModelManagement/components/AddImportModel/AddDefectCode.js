import React from 'react'
import { injectIntl } from 'react-intl'
import { Form, Input, Icon, Button, Checkbox, Modal, InputNumber } from 'antd'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import classes from './AddImportModel.scss'

type Props = {
  that: Object,
  form: Object,
  intl: Object,
}

const FormItem = Form.Item
class DynamicFieldSet extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      uuid: this.props.that.uuid,
    }
  }
  componentWillReceiveProps() {
    if (this.props.that.state.visible === false) {
      this.props.form.resetFields()
    }
  }

  // 删除一个defectCode
  remove = (k) => {
    const { form } = this.props
    // can use data-binding to get
    const keys = form.getFieldValue('keys')
    // We need at least one passenger
    if (keys.length === 1) {
      return
    }

    // can use data-binding to set
    form.setFieldsValue({
      keys: keys.filter(key => key !== k),
    })
  }

  // 添加一个defectCode
  add = () => {
    // uuid = uuid + 1
    this.setState({
      uuid: this.state.uuid + 1,
    }, () => {
      const { form } = this.props
      // can use data-binding to get
      const keys = form.getFieldValue('keys')

      const nextKeys = keys.concat(this.state.uuid)
      let newNextKeys = []
      newNextKeys = nextKeys.map((index) => {
        return newNextKeys.push(index)
      })
      // can use data-binding to set
      // important! notify form to detect changes
      form.setFieldsValue({
        keys: newNextKeys,
      })
    })
  }

  props: Props
  render() {
    const { intl } = this.props
    const { getFieldDecorator, getFieldValue } = this.props.form
    getFieldDecorator('keys', { initialValue: [1] })
    const keys = getFieldValue('keys')
    // formItems 代表defectCode 可以添加 可以删除
    const formItems = keys.map((k, index) => {
      return (
        <FormItem
          required={false}
          key={k}
        >
          <div style={{ display: 'flex', justifyContent: 'start', margin: '10px 10px 0 10px' }}>
            <WrapperFormattedMessage
              id="modelManagement.addImportModel.defectCodeName"
            />{index + 1}：
            {getFieldDecorator(`names_${k}`, {
            })(
              <Input
                placeholder={intl.formatMessage({ id: 'modelManagement.addImportModel.defectCodeName' })}
                style={{ width: '40%', marginRight: 8 }}
              />,
            )}
            <WrapperFormattedMessage
              id="modelManagement.addImportModel.threshold"
            />
            {getFieldDecorator(`value_${k}`, {
            })(
              <InputNumber
                min={0}
                max={100}
                placeholder={intl.formatMessage({ id: 'modelManagement.addImportModel.threshold' })}
                style={{ width: '20%', marginRight: 8 }}
              />,
            )}
            {keys.length > 1 ? (
              <Icon
                className="dynamic-delete-button"
                type="minus-circle-o"
                disabled={keys.length === 1}
                onClick={() => this.remove(k)}
              />
            ) : null}
          </div>
        </FormItem>
      )
    })
    return (
      <Modal
        title={intl.formatMessage({ id: 'modelManagement.addImportModel.importModel' })}
        visible={this.props.that.state.visible}
        onCancel={this.props.that.handleCancel}
        onOk={this.props.that.handleOk(intl)}
      >
        <Form>
          <FormItem>
            <div className={classes.filter}>
              <WrapperFormattedMessage
                id="modelName"
              />
              <input
                type='text'
                className={'pt-input'}
                value={this.props.that.state.modelName}
                onChange={this.props.that.inputChange('modelName')}
              />
            </div>
          </FormItem>
          <FormItem>
            <div className={classes.filter}>
              <WrapperFormattedMessage
                id="description"
              />
              <textarea
                style={{ width: '200px' }}
                className={'pt-input'}
                value={this.props.that.state.discription}
                onChange={this.props.that.inputChange('discription')}
              />
            </div>
          </FormItem>
          <FormItem>
            <div className={classes.filter}>
              <WrapperFormattedMessage
                id="modelManagement.addImportModel.path"
              />
              <input
                type='text'
                className={'pt-input'}
                value={this.props.that.state.imagePath}
                onChange={this.props.that.inputChange('imagePath')}
              />
            </div>
          </FormItem>
          <FormItem>
            <div className={classes.filter} style={{ justifyContent: 'start' }}>
              <WrapperFormattedMessage
                id="modelManagement.addImportModel.defaultModel"
              />
              <Checkbox
                style={{ width: '200px', marginLeft: '50px' }}
                checked={this.props.that.state.defaultModel}
                onChange={this.props.that.checkboxDefault}
              ><WrapperFormattedMessage
                id="modelManagement.addImportModel.setAsDefault"
              /></Checkbox>
            </div>
          </FormItem>
          <div style={{ height: '150px', overflowY: 'scroll' }}>
            {formItems}
            <Button style={{ marginLeft: '380px', marginRight: '10px' }} type="primary" shape="circle" icon="plus" onClick={this.add} />
          </div>
        </Form>
      </Modal>
    )
  }
}

const AddDefectCode = Form.create()(DynamicFieldSet)
export default injectIntl(AddDefectCode)
