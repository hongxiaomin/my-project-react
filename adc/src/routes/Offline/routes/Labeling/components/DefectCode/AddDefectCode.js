import React from 'react'
import { Modal, Button, Form, Input } from 'antd'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { addDefectCode } from '../../modules'

const FormItem = Form.Item
export class AddDefectCode extends React.Component {
  constructor() {
    super()

    this.state = {
      loading: false,
    }
  }

  handleOk = (e) => {
    const { resetFields } = this.props.form
    e.preventDefault()
    this.props.form.validateFieldsAndScroll((err, values) => {
      if (err) return
      this.setState({
        loading: true,
      })
      this.props.dispatch(addDefectCode(values.defectCode.toUpperCase(), () => {
        this.setState({
          loading: false,
        })
        resetFields()
        this.props.handleCancel()
      }))
    })
  }

  render() {
    const { intl } = this.props
    const { getFieldDecorator } = this.props.form
    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 6 },
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 14 },
      },
    }
    return (<Modal
      visible={this.props.visible}
      title={intl.formatMessage({ id: 'labelingTool.defectCode.addNewDefectCode' })}
      onOk={this.handleOk}
      onCancel={this.props.handleCancel}
      footer={[
        <Button key="back" size="large" onClick={this.props.handleCancel}>
          <WrapperFormattedMessage id="cancel" />
        </Button>,
        <Button key="submit" type="primary" size="large" loading={this.state.loading} onClick={this.handleOk}>
          <WrapperFormattedMessage id="Save" />
        </Button>,
      ]}
    >
      <Form>
        <FormItem
          label={intl.formatMessage({ id: 'labelingTool.defectCode.defectCode' })}
          {...formItemLayout}
        >
          {getFieldDecorator('defectCode', {
            rules: [{
              required: true, message: intl.formatMessage({ id: 'labelingTool.defectCode.defectCodeError' }),
            }],
          })(<Input />)}
        </FormItem>
      </Form>
    </Modal>)
  }
}

AddDefectCode.propTypes = {
  visible: PropTypes.bool.isRequired,
  form: PropTypes.object.isRequired,
  handleCancel: PropTypes.func.isRequired,
  dispatch: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
}

const DefectCodeForm = Form.create()(injectIntl(AddDefectCode))
export default connect()(DefectCodeForm)
