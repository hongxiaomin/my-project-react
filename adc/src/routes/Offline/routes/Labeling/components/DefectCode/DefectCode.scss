.defect-code {
  border: 1px solid #979797;
  min-height: 500px;

  h4 {
    background: #4A90E2;
    color: #fff;
    font-size: 14px;
    padding: 10px;


    &:after {
      content: '';
      display: block;
      clear: both;
    }
  }
}
