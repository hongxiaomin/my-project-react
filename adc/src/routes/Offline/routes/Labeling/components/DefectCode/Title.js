import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import { Icon } from 'antd'
import DataList from './DataList'
import DefectCodeForm from './AddDefectCode'
import { updateResetKeyCodeStatus, UPDATE_DEFAULT_DEFECT_CODE_LIST } from '../../modules'

export class Title extends React.Component {
  constructor() {
    super()

    this.state = {
      visible: false,
    }
  }

  toggle = () => {
    this.handleFocus()
    this.setState({
      visible: !this.state.visible,
    })
  }

  handleBlur = () => {
    this.props.dispatch(updateResetKeyCodeStatus(false))
  }

  handleFocus = () => {
    this.props.dispatch(updateResetKeyCodeStatus(true))
  }

  handleChange = (defect, cb) => {
    const { defaultDefectCodeList, dispatch } = this.props
    if (!defaultDefectCodeList.some(item => item.defectId === defect.defectId)) {
      dispatch({
        type: UPDATE_DEFAULT_DEFECT_CODE_LIST,
        data: defaultDefectCodeList.concat(defect),
      })
    }
    if (cb) cb()
  }

  handleCancel = () => {
    this.handleBlur()
    this.setState({
      visible: false,
    })
  }

  render() {
    const { defectCodeList, defaultDefectCodeList, intl } = this.props

    return (<div>
      <h4><WrapperFormattedMessage id="labelingTool.defectCode.selectDefectCode" /> <Icon
        alt={intl.formatMessage({ id: 'labelingTool.defectCode.addDefect' })}
        title={intl.formatMessage({ id: 'labelingTool.defectCode.addDefect' })}
        type="plus-circle-o"
        style={{ fontSize: 16, float: 'right', cursor: 'pointer' }}
        onClick={this.toggle}
      />
      </h4>
      <DataList
        data={defectCodeList}
        selectedData={defaultDefectCodeList}
        onChange={this.handleChange}
        onBlur={this.handleBlur}
        onFocus={this.handleFocus}
      />

      <DefectCodeForm
        visible={this.state.visible}
        handleCancel={this.handleCancel}
      />
    </div>)
  }
}

Title.propTypes = {
  defectCodeList: PropTypes.array.isRequired,
  defaultDefectCodeList: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(({ labelingTool, intl: { locale } }) => ({
  locale,
  defectCodeList: labelingTool.defectCodeList,
  defaultDefectCodeList: labelingTool.defaultDefectCodeList,
}))(injectIntl(Title))
