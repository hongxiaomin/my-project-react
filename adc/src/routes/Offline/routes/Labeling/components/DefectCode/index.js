import DefectCodeContainer from './DefectCode'

export default DefectCodeContainer
