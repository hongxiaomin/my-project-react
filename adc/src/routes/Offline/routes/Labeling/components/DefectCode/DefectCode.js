import React from 'react'
import { connect } from 'react-redux'
import { injectIntl } from 'react-intl'
import PropTypes from 'prop-types'
import { Icon } from 'antd'
import ReactTable from 'components/ADCTable'
import Title from './Title'
import './DefectCode.scss'
import { getSelectInfo, UPDATE_DEFAULT_DEFECT_CODE_LIST, updateSelectedDefaultDefect } from '../../modules'


export class DefectCode extends React.Component {
  constructor() {
    super()
    this.state = {
      index: undefined,
    }
  }

  componentDidMount() {
    this.props.dispatch(getSelectInfo())
  }

  handleTrClick = (data) => {
    this.props.dispatch(updateSelectedDefaultDefect(data))
  }

  del = ({ original }) => (e) => {
    const { dispatch, defaultDefectCodeList } = this.props
    dispatch({
      type: UPDATE_DEFAULT_DEFECT_CODE_LIST,
      data: defaultDefectCodeList.filter(item => item.defectId !== original.defectId),
    })
    e.stopPropagation()
  }

  handleTrMouseOver = (rowInfo) => {
    this.setState({
      index: rowInfo.index,
    })
  }

  handleTrMouseLeave = () => {
    this.setState({
      index: undefined,
    })
  }

  render() {
    const { defaultDefectCodeList, selectedDefaultDefect, intl } = this.props
    return (<div styleName="defect-code">
      <Title />
      <ReactTable
        pageSize={14}
        showPagination={false}
        data={defaultDefectCodeList}
        getTrProps={(state, rowInfo) => {
          const style = { textAlign: 'center', height: '35px' }
          if (rowInfo && rowInfo.original.defectCode === selectedDefaultDefect.defectCode) {
            style.background = '#CAE6F5'
          }
          return {
            onClick: () => {
              this.handleTrClick(rowInfo.original)
            },
            onMouseEnter: () => {
              this.handleTrMouseOver(rowInfo)
            },
            onMouseLeave: this.handleTrMouseLeave,
            style,
          }
        }}
        columns={[
          {
            Header: intl.formatMessage({ id: 'labelingTool.defectCode.defectCode' }),
            accessor: 'defectCode',
            Cell: row => (<div style={{ position: 'relative', cursor: 'pointer' }}>
              <span
                title={row.value}
                style={{
                  wordWrap: 'break-word',
                }}
              >
                {row.value}
              </span>
              <Icon
                alt={`${intl.formatMessage({ id: 'delete' })} ${row.value}`}
                title={`${intl.formatMessage({ id: 'delete' })} ${row.value}`}
                type="close-circle"
                style={{
                  float: 'right',
                  position: 'absolute',
                  right: 3,
                  top: 3,
                  color: '#D0011B',
                  background: row.original.defectCode === selectedDefaultDefect.defectCode ? '#CAE6F5' : '#fff',
                  cursor: 'pointer',
                  display: this.state.index === row.index ? '' : 'none',
                }}
                onClick={this.del(row)}
              />
            </div>),
          },
        ]}
        style={{
          height: '555px',
        }}
      />
    </div>)
  }
}

DefectCode.propTypes = {
  dispatch: PropTypes.func.isRequired,
  defaultDefectCodeList: PropTypes.array.isRequired,
  selectedDefaultDefect: PropTypes.object.isRequired,
  intl: PropTypes.object.isRequired,
}

export default connect(({ labelingTool, intl: { locale } }) => ({
  locale,
  defaultDefectCodeList: labelingTool.defaultDefectCodeList,
  selectedDefaultDefect: labelingTool.selectedDefaultDefect,
}))(injectIntl(DefectCode))
