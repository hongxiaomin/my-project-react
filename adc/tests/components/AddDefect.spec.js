import React from 'react'
import { shallow, mount } from 'enzyme'
import _ from 'lodash'
import { AddDefectCode } from 'routes/Offline/routes/Labeling/components/DefectCode/AddDefectCode'

describe('(Component) AddDefectCode', () => {
  const visible = true
  const handleCancel = jest.fn()
  const addDefectCode = jest.fn()
  it('render normally', () => {
    const wrapper = shallow(
      <AddDefectCode
        visible={visible}
        form={{
          setFieldsValue: () => {},
          getFieldsValue: () => {},
          getFieldDecorator: () => () => {},
          setFields: () => {},
        }}
        handleCancel={handleCancel}
        addDefectCode={addDefectCode}
      />,
    )
    // const nativeEvent = { nativeEvent: { stopImmediatePropagation: _.noop } }
    // wrapper.find('Modal').simulate('click', nativeEvent)
    // expect(wrapper.state('loading')).toBe(false)
    expect(wrapper).toMatchSnapshot()
  })
})

