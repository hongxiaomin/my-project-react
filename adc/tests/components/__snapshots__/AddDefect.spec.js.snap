// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) AddDefectCode render normally 1`] = `
<Modal
  confirmLoading={false}
  footer={
    Array [
      <Button
        clicked={false}
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        size="large"
    >
        Cancel
    </Button>,
      <Button
        clicked={false}
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        size="large"
        type="primary"
    >
        Save
    </Button>,
    ]
  }
  maskTransitionName="fade"
  onCancel={[Function]}
  onOk={[Function]}
  prefixCls="ant-modal"
  title="Add New Defect Code"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Form
    hideRequiredMark={false}
    layout="horizontal"
    onSubmit={[Function]}
    prefixCls="ant-form"
  >
    <FormItem
      colon={true}
      hasFeedback={false}
      label="Defect Code"
      labelCol={
        Object {
          "sm": Object {
            "span": 6,
          },
          "xs": Object {
            "span": 24,
          },
        }
      }
      prefixCls="ant-form"
      wrapperCol={
        Object {
          "sm": Object {
            "span": 14,
          },
          "xs": Object {
            "span": 24,
          },
        }
      }
    />
  </Form>
</Modal>
`;
